### DISCUSSION TITLE
Do ShapeCrawler support password protection on individual slide or entire presentation

### DISCUSSION THREAD

**Discussion by ZubairImran** [ShapeCrawler/ShapeCrawler#791] [Ideas]
Hi @ashahabov ,
would like to know the support of password protection of individual slide or entire presentation is available with the current library

**Comment by AdamShakhabov** [comment]
Hello @ZubairImran,

No, it is not supported yet.

**Reply by ZubairImran** [reply]
Thanks @ashahabov for the prompt reply. Any plan to support in near future?

**Reply by AdamShakhabov** [reply]
Yes, it is planned. No ETA yet.

I've already tried to make a quick implementation, but it is not so easy. My quick research states that the encryption of Open XML document requires some [[MS_OFFCRYPTO]](https://learn.microsoft.com/en-us/openspecs/office_file_formats/ms-offcrypto/24e24b4b-1c15-4229-944a-ffcb8c0b2dae).

**Reply by ZubairImran** [reply]
Hi @ashahabov 
Thanks for your effort and details. Will follow up https://github.com/ShapeCrawler/ShapeCrawler/issues/792
